### ISSUE TITLE
Any face landmark detection example with Dali?

### ISSUE THREAD

**Issue by xjcvip007** [NVIDIA/DALI#1191]
Thanks for your amazing job~
I want to use Dali instead of dataloader to help our face landmark detection with pytorch.
I construct a csv with img_name and 106 facial landmark points as shown below:
`0. jpg, 13.21314, 12,342, 42.3242, 54.23123,...`
So I use ExternalInputIterator and ExternalSourcePipeline to build a pipeline. But I have a question,
how can I modify my landmarks with different augmentation in our pipeline. For example, if I do resize augment operator, I need to resize the image and scale landmarks together. Or I do rotate operator, I need to rotate the image and rotate landmarks together. How can I do it in the ExternalSourcePipeline? Below is my code:

`
class ExternalInputIterator(object):

    def __init__(self, batch_size, root_dir, img_dir, csv_file):
        self.img_dir = os.path.join(root_dir, img_dir)
        self.batch_size = batch_size
        landmark_file_name = pd.read_csv(os.path.join(root_dir, csv_file), header=None)
        self.img_names = landmark_file_name.loc[:, 0].values
        self.landmarks = landmark_file_name.loc[:, 1: 2*106].values.astype('float')

        #    self.files = [line.rstrip() for line in f if line is not '']
        rnd_index = list(range(self.img_names.shape[0]))
        shuffle(rnd_index)
        self.img_names = self.img_names[rnd_index]
        self.landmarks = self.landmarks[rnd_index]

    def __iter__(self):
        self.i = 0
        self.n = self.img_names.shape[0]
        return self

    def __next__(self):
        batch = []
        lmks = []
        for _ in range(self.batch_size):
            img_name, landmark = self.img_names[self.i], self.landmarks[self.i]
            f = open(os.path.join(self.img_dir, img_name), 'rb')
            batch.append(np.frombuffer(f.read(), dtype = np.uint8))
            lmks.append(landmark)
            self.i = (self.i + 1) % self.n
        return (batch, lmks)

    next = __next__

class ExternalSourcePipeline(Pipeline):

    def __init__(self, batch_size, num_threads, device_id):
        super(ExternalSourcePipeline, self).__init__(batch_size,
                                      num_threads,
                                      device_id,
                                      seed=12)
        self.input = ops.ExternalSource()
        self.input_label = ops.ExternalSource()
        self.decode = ops.ImageDecoder(device = "mixed", output_type = types.RGB)
        self.cast = ops.Cast(device = "gpu",
                             dtype = types.INT32)

    def define_graph(self):
        self.jpegs = self.input()
        self.labels = self.input_label()
        images = self.decode(self.jpegs)
        output = self.cast(images)
        print(self.input_label, self.labels)
        return (output, self.labels)

    def iter_setup(self):
        (images, labels) = iterator.next()
        print(images)
        self.feed_input(self.jpegs, images)
        self.feed_input(self.labels, labels)

**Comment by awolant** [comment]
Hi, thanks for the question.
Right now DALI does not support facial landmarks, so there is no existing way of doing this.
You need your own custom operators to process landmarks. 
You can implement you own operators and use them in DALI pipeline using our [plugin system](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/examples/extend/create_a_custom_operator.html).
If you would rather start with something easier you can look into [PythonFunction operator](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/examples/python_operator.html?highlight=python%20function) to prototype. Please keep in mind that `PythonFunction` delivers inferior performance and is a tool to prototype/debug.`

**Comment by JanuszL** [comment]
Hi,
[This example](https://docs.nvidia.com/deeplearning/dali/user-guide/docs/examples/math/geometric_transforms.html) should provide you the required reference.
